apis/beacon/blobs/blobs.yaml

```diff
@@ -0,0 +1,62 @@
+get:
+  operationId: getBlobs
+  summary: Get blobs
+  description: |
+    Retrieves blobs for a given block id.
+    Depending on `Accept` header it can be returned either as json or as bytes serialized by SSZ.
+    It is recommended that blob retrieval for finalized blocks should by done by slot else the underlying index might not be available.
+  tags:
+    - Beacon
+  parameters:
+    - name: block_id
+      in: path
+      required: true
+      $ref: '../../../beacon-node-oapi.yaml#/components/parameters/BlockId'
+    - name: indices
+      in: query
+      description: Array of indices for blobs to request for in the specified block. Returns all blobs in the block if not speicfied.
+      required: false
+      schema:
+        type: array
+        uniqueItems: true
+        items:
+          $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Uint64'
+  responses:
+    "200":
+      description: "Successful response"
+      content:
+        application/json:
+          schema:
+            title: GetBlobsResponse
+            type: object
+            properties:
+              data:
+                type: array
+                items:
+                  oneOf:
+                    - $ref: "../../../beacon-node-oapi.yaml#/components/schemas/Deneb.BlobSidecar"
+                minItems: 0
+                maxItems: 4
+        application/octet-stream:
+          schema:
+            description: "SSZ serialized block bytes. Use Accept header to choose this response type"
+    "400":
+      description: "The block ID supplied could not be parsed"
+      content:
+        application/json:
+          schema:
+            $ref: "../../../beacon-node-oapi.yaml#/components/schemas/ErrorMessage"
+            example:
+              code: 400
+              message: "Invalid block ID: current"
+    "404":
+      description: "Blob sidecar not found"
+      content:
+        application/json:
+          schema:
+            $ref: "../../../beacon-node-oapi.yaml#/components/schemas/ErrorMessage"
+            example:
+              code: 404
+              message: "Blob sidecar not found"
```

Comment:
this is fairly reversed...

given the block id is in the path, then i'd expect a 404 to be the block itself not found...

400 would potentially be better for specific blobs not found?

Would a partial availability still be not found for blobs? so if i request `indicies=0,1`, and blob 1 is missing, do i return a 400?

Reply:
i think if any of the indices is missing, the full api should fail with 400 and may be error indicating the missed index

Reply:
I realised, its effectively an array return and what we generally do is return an empty array, not an error... so unless we need an error to indicate something was physically missing, following the pattern of empty array means nothing would likely be preferable from a consistency point of view...

Reply:
Agree with the above on consistency - and only return blobs matching the indices, e.g. `indices=1` returns empty array if blob 1 is missing.

if the block itself is not found, then a 404  is returned like @rolfyone suggested. I'll update the message there.

Reply:
Updated 404 description to "Block not found" in https://github.com/ethereum/beacon-APIs/pull/286/commits/4ccaaa6f018c596c6ebf1227943da952ed7cf29f